compiler/modules/front.syntax/ast.bal

```diff
@@ -231,6 +231,12 @@ public type TypeTestExpr record {|
     t:SemType semType;
 |};
 
+public type TypeTestNotExpr record {|
```

Comment:
What is this?

Reply:
This to represent expressions with `!is` in the ast and differentiate from `is` expressions

Reply:
should I use the same `TypeTestExpr` for both with a new field to indicate `!is` ?

Reply:
You already added the `negated` field to TypeTestExpr for this...

Reply:
I only added negated for the TypeTestExpr in the bir but not the one in the ast. I can do the same to both to keep consistency

Reply:
Yes...

Reply:
fixed